### PR TITLE
#164857208 Setup  base application for heroku

### PR DESCRIPTION
**Pull request by bencyn**
**What does this PR do?**

Sets up a base application with Heroku configurations to allow deployment

**Tasks to be completed**

- [x] Add a procfile with Heroku configurations 
- [x] Install heroku dependencies

**Heroku deployment links**
[staging environment](https://ah-premier-staging.herokuapp.com/)
[production environment](https://ah-premier.herokuapp.com/)

**How can this be manually tested**

- Clone the repo and checkout the `ch/164857208-deploy-to-heroku` branch

- Create and activate a virtual environment
```
$ virtualenv env
$ source env/bin/activate
```

- Install requirements
```
$ pip install -r requirements.txt
```

- Make migrations
```
$ python manage.py makemigrations
```

- Run migrations

```
$ python manage.py migrate
```

- Run the tests
```
python manage.py test
```

**Relevant Pivotal tracker stories** 
[#164857208](https://www.pivotaltracker.com/story/show/164857208)

**Screenshots**
N/A

### CODE REVIEW

**Comment by Tevinthuku** [comment]
@bencyn what's the `.requirements.txt.swp` for?

**Comment by Tevinthuku** [comment]
You also have a linting issue 
image

File: Procfile [hunk 1]
```diff
@@ -0,0 +1 @@
+web: gunicorn djangorest.wsgi
```
Comment (Tevinthuku) [inline Procfile]: :thinking:  I think this should be `authors.wsgi`

**Review by Tevinthuku — COMMENTED** [review]
it needs to be authors.wsgi because the wsgi.py is in the authors directory.

**Comment by bencyn** [comment]
> it needs to be authors.wsgi because the wsgi.py is in the authors directory.

File: Procfile (hunk 1 above)
Comment (bencyn) [reply Procfile]: I've updated accordingly thanks for pointing out that

Comment (Tevinthuku) [reply Procfile]: :+1: Now you can work on indentation and linting issues. Are you using flake8?

Comment (Tevinthuku) [reply Procfile]: There's this one issue
image
You can install flake8 and autopep8
are you using vs-code?

**Review by martinMutuma — APPROVED** [review]
Code added looks good
A rebase is required before merge

**Comment by bencyn** [comment]
@martinMutuma thanks for the input I've rebased accordingly.

**Review by Tevinthuku — COMMENTED** [review]
Almost forgot this. As the app stands Heroku will not do any migrations from its end. Kindly add a release command that will run the migration
https://devcenter.heroku.com/articles/release-phase#specifying-release-phase-tasks

**Comment by Tevinthuku** [comment]
> **What does this PR do?**
> 
> Sets up a base application with Heroku configurations to allow deployment
> 
> **Tasks to be completed**
> 
> * [x]  Add a procfile with Heroku configurations
> * [x]  Install heroku dependencies
> 
> **Heroku deployment links**
> [staging environment](https://ah-premier-staging.herokuapp.com/)
> [production environment](https://ah-premier.herokuapp.com/)
> 
> **How can this be manually tested**
> 
> * Clone the repo and checkout the `ch/164857208-deploy-to-heroku` branch
> * Create and activate a virtual environment
> 
> ```
> $ virtualenv env
> $ source env/bin/activate
> ```
> * Install requirements
> 
> ```
> $ pip install -r requirements.txt
> ```
> * Make migrations
> 
> ```
> $ python manage.py makemigrations
> ```
> * Run migrations
> 
> ```
> $ python manage.py migrate
> ```
> * Run the tests
> 
> ```
> python manage.py test
> ```
> **Relevant Pivotal tracker stories**
> [#164857208](https://www.pivotaltracker.com/story/show/164857208)
> 
> **Screenshots**
> N/A


I like the fact that we willl be having two deployments :) Its a nice feature

**Review by Tevinthuku — APPROVED** [review]
:+1:  Looks good now

File: .gitignore [hunk 2]
```diff
@@ -91,3 +91,5 @@ ENV/
 
 # SQLite3
 db.sqlite3
+
+.requirements.txt.swp
```
Comment (miritih) [inline .gitignore]: Add a blank line at the end of the file. 
```suggestion
.requirements.txt.swp

```
It's advisable to have files generated by your system such as `swp files` ignored inside your `global git ignore file`
here is a resource on how to add a global gitignore file. 
https://help.github.com/en/articles/ignoring-files

File: Procfile [hunk 3]
```diff
@@ -0,0 +1,2 @@
+release: python manage.py migrate
+web: gunicorn authors.wsgi
```
Comment (miritih) [inline Procfile]: Blank line at the end of file
```suggestion
web: gunicorn authors.wsgi

```

**Review by miritih — CHANGES_REQUESTED** [review]

**Review by miritih — APPROVED** [review]

**Pull request merged**